homeassistant/components/komfovent/climate.py

```diff
@@ -0,0 +1,84 @@
+"""Ventilation Units from Komfovent integration."""
+from __future__ import annotations
+
+import komfovent_api
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, MANUFACTURER, PARAM_HOST, PARAM_PASSWORD, PARAM_USERNAME
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Komfovent unit control."""
+    conf_host = str(entry.data[PARAM_HOST])
+    conf_username = str(entry.data[PARAM_USERNAME])
+    conf_password = str(entry.data[PARAM_PASSWORD])
+    _, credentials = komfovent_api.get_credentials(
+        conf_host, conf_username, conf_password
+    )
+
+    result, settings = await komfovent_api.get_settings(credentials)
+    if result == komfovent_api.KomfoventConnectionResult.SUCCESS:
+        async_add_entities([KomfoventDevice(credentials, settings)], True)
+
+
+class KomfoventDevice(ClimateEntity):
+    """Representation of a ventilation unit."""
+
+    _attr_hvac_modes = [HVACMode.FAN_ONLY]
+    _attr_hvac_mode = HVACMode.FAN_ONLY
```

Comment:
It is a ventilation unit. So the closest hvac mode i could find is a fan. HASS (from my testing) currently doesnt support not setting and HVAC mode on climate entity, it's not supported by the code and breaks the FE controls

Reply:
It is a ventilation unit. So the closest hvac mode i could find is a fan. HASS (from my testing) currently doesnt support not setting and HVAC mode on climate entity, it's not supported by the code and breaks the FE controls

Reply:
> It is a ventilation unit

It should be using the `fan` entity, not the `climate` entity component.

HVAC isn't a factor in this, as the only mode available is `FAN_ONLY`, which is the sole purpose of the fan platform.

Please adjust the PR to use the `fan` entity component instead.

Reply:
It's not a fan though, it supports heating or not heating the air internally, reporting and setting target temperature as well as moisture level control. I am just not implementing them in this MR for the sake of simplicity. TO me it seems much closer to a climate than a fan device.

Reply:
>  it supports heating or not heating the air internally, reporting and setting target temperature

That should be implemented in this PR, otherwise this PR isn't valid.